src/components/ChataTable/ChataTable.js

```diff
@@ -1068,11 +1068,18 @@ export default class ChataTable extends React.Component {
     this.setState({ contextMenuColumn: undefined })
 
     const currentAdditionalSelectColumns = this.props.response?.data?.data?.fe_req?.additional_selects ?? []
-    const newAdditionalSelectColumns = currentAdditionalSelectColumns?.filter(
-      (select) => {
-        return select?.columns[0]?.replace(/ /g, '') !== column?.name?.replace(/ /g, '')
-      },
-    )
+    const newAdditionalSelectColumns = currentAdditionalSelectColumns?.map((select) => {
+      if (
+        select.columns[0] !== column.name ||
+        select?.columns[0]?.replace(/ /g, '') !== column?.name?.replace(/ /g, '')
+      ) {
+        return select
+      } else if (select.columns[0] === column.name) {
```

Comment:
I think the .filter() still needs to be here, as `newAdditionalSelectColumns` get's passed into an endpoint that updates the query response and the .map() func doesn't remove items.

What is the goal here? The columns were being removed correctly but then were not showing up in the add column list?

Reply:
I wasn't able to use filter here. The filter was causing the column name to be removed from the list after it was being removed. Yes column was being removed, but was not in the add column list

Reply:
We need to filter here because we need to update the queryOutput at the sql level. By removing it from the additional_selects we update the query at the SQL level. 

If we check the query endpoint after removing the column we should see the removed column not included in the `additional_selects` lists of the payload, but then it should be in the response in the `available_selects` list. If it is in the available_selects list then something isn't updating properly, probably something in the AddColumnBtn component, if it isn't in the list then i think there might a bug in the BE.

Reply:
i think the filter / map are just creating new array of values, which is still being saved to the variable. I changed because the filter is removing the option from the newAdditionalSelectColumns, vs hiding is using the visible flags. I think we should be using the following to hide the columns instead
column.visible = false
column.is_visible = false